pkg/controllers/deprovisioning/expiration.go

```diff
@@ -74,7 +70,7 @@ func (e *Expiration) filterAndSortCandidates(ctx context.Context, nodes []*Candi
 		return nil, fmt.Errorf("filtering candidates, %w", err)
 	}
 	sort.Slice(candidates, func(i int, j int) bool {
-		return node.GetExpirationTime(candidates[i].Node, candidates[i].provisioner).Before(node.GetExpirationTime(candidates[j].Node, candidates[j].provisioner))
+		return functional.GetExpirationTime(candidates[i].Node, candidates[i].provisioner).Before(functional.GetExpirationTime(candidates[j].Node, candidates[j].provisioner))
```

Comment:
Just a thought, as I read this. You could sort nodes by age, rather than time expired. This still makes semantic sense with expiration, with the benefit of being a generalizable rule that could apply broadly to all deprovisioning methods.

Reply:
Yep, I was thinking the same thing. I wanted to package it in a separate PR, but I was thinking that we would sort by age 👍

Reply:
Is there any notion of different nodes in a cluster having different expiration configuration?

For example, if there is a node that is 10 days old, but w/ a TTL of 30 days, and a node that is 3 days old, w/ a 24 hour TTL, we wouldn't want to prioritize the older node, right?

Reply:
What about sorting it by how long it's been disrupted? This works for all disruption mechanisms I believe.

Reply:
+1, and you get it for free with transition timestamp on status conditions.